src/tests/ov_helpers/ov_snippets_models/include/subgraph_mha.hpp

```diff
@@ -443,6 +443,36 @@ class MHAWithExtractedReshapeFunction : public SnippetsFunctionBase {
     bool add_2nd_reshape = false;
 };
 
+/* Graph:
+ *           input0   input1
+ *              \     /
+ *              MatMul0  input2
+ *                 |     /
+ *              Eltwise1
+ *                 |
+ *              Reshape input3
+ *                 |    /
+ *              Eltwise2
+ *                 |
+ *              Reshape
+ *                 |
+ *              Softmax
+ *                 |       input4
+ *                  \      /
+ *                   MatMul1
+ */
+class MHARankUpgradeToReductionFunction : public SnippetsFunctionBase {
+public:
+    explicit MHARankUpgradeToReductionFunction(const std::vector<PartialShape>& inputShapes)
+        : SnippetsFunctionBase(inputShapes) {
+        OPENVINO_ASSERT(input_shapes.size() == 5, "Got invalid number of input shapes");
```

Comment:
Can we add a check that all inputs shapes must be static?

Reply:
Sure, have added. thanks Vladislav!